### ISSUE TITLE
strftime on unicode Python error on run

### ISSUE THREAD

**Issue by okapia** [untitaker/vdirsyncer#275]
I've just newly installed vdirsyncer on FreeBSD from the prebuild packages so this is using python 2.7. When I run a sync, I get the following error:
```
Unhandled exception occured for oliver.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/vdirsyncer/cli/tasks.py", line 72, in sync_collection
    force_delete=force_delete
  File "/usr/local/lib/python2.7/site-packages/vdirsyncer/sync.py", line 158, in sync
    b_idents = _prepare_idents(storage_b, storage_a, b_href_to_status)
  File "/usr/local/lib/python2.7/site-packages/vdirsyncer/sync.py", line 113, in _prepare_idents
    hrefs = _prepare_hrefs(storage, other_storage, href_to_status)
  File "/usr/local/lib/python2.7/site-packages/vdirsyncer/sync.py", line 98, in _prepare_hrefs
    for href, etag in storage.list():
  File "/usr/local/lib/python2.7/site-packages/vdirsyncer/storage/dav.py", line 633, in list
    for caldavfilter in caldavfilters:
  File "/usr/local/lib/python2.7/site-packages/vdirsyncer/storage/dav.py", line 605, in _get_list_filters
    start, end):
  File "/usr/local/lib/python2.7/site-packages/vdirsyncer/storage/dav.py", line 591, in _get_list_filters
    start = start.strftime(CALDAV_DT_FORMAT)
AttributeError: 'unicode' object has no attribute 'strftime'
```

I'm guessing that start should be some sort of date object rather than of type unicode. Any idea what the cause of this might be? Thanks

**Comment by untitaker** [comment]
Could you post your configuration for that storage?

**Comment by okapia** [comment]
Seems the problem is related to these two lines copied from the website:
```
start_date = datetime.now() - timedelta(days=365)
end_date = datetime.now() + timedelta(days=365)
```
Full config is as follows. The intention is to sync between local storage for khal and owncloud.
```
[general]
status_path = /home/opk/.cache/vdirsyncer
[pair opk]
a = opk_local
b = opk_cloud

[storage opk_local]
type = filesystem
path = ~/.calendars/opk/
fileext = .ics

[storage opk_cloud]
type = caldav
url = "https://192.168.178.23/remote.php/calendarplus/calendars/opk/opk"
start_date = datetime.now() - timedelta(days=365)
end_date = datetime.now() + timedelta(days=365)
username = "opk"
verify_fingerprint = "F0:10:AD:EF:36:EF:9E:3E:81:A9:C5:3D:F2:71:76:40:30:4C:2A:1F"
```

**Comment by untitaker** [comment]
These lines work fine for me. Which version of vdirsyncer are you using?

**Comment by okapia** [comment]
0.4.3

I hadn't realised how old the vdirsyncer in FreeBSD's ports was. It is Python 2.7 if that has any impact on it.

**Comment by untitaker** [comment]
It should work fine with that version too, but according to [pkgsrc](http://pkgsrc.se/time/py-vdirsyncer) the latest version is 0.6.0...

**Comment by geier** [comment]
FreeBSD doesn't use pkg source though.

**Comment by untitaker** [comment]
@okapia Have you tried upgrading vdirsyncer?

**Comment by untitaker** [comment]
@okapia any updates on this?

**Comment by okapia** [comment]
Sorry, I had not upgraded yet because I was lazily hoping that it wouldn't be long before pkg update would upgrade it. It is still out-of-date there. I was also assuming that a manual upgrade would entail manual installs or upgrades of a whole load of python dependencies which discouraged me from trying.

Anyway, I've just tested a manual install of vdirsyncer from git and can confirm that the problem is gone. I had to make changes to the config file but that was essentially changes to reflect documented changes in vdirsyncer.

Thanks for your help and sorry for reporting the issue without first upgrading.
